### PR TITLE
feat: console warning added when using an invalid API key

### LINKED ISSUE #1068: Error handling in setApiKey
### Issue Summary
I'm retrieving the api key from an environment variable. Somehow in one of our environments the api key was `undefined`. It would be helpful if an error is logged or an error is thrown when setupApiKey is called with `undefined`, or with other wrong inputs, so that we are immediately informed when this is the case.

### Steps to Reproduce
1. Setup the api key as undefined `sgMail.setupApiKey(undefined)`
2. No error, so it seems like the ApiKey is set correctly
3. Only when trying to send a mail an error is thrown

### Code Snippet
```node
const msg = {
  to: 'test@example.com',
  from: 'test@example.com',
  subject: 'Sending with SendGrid is Fun',
  text: 'and easy to do anywhere, even with Node.js',
  html: '<strong>and easy to do anywhere, even with Node.js</strong>',
};
sgMail.setApiKey(undefined);
sgMail.send(msg);
```

### Exception/Log

None after `setupApiKey(undefined)`

Error after sending the mail;
```
Unauthorized (401)
  Permission denied, wrong credentials
```

### Technical details:
* sendgrid-nodejs version: 6.5.4
* node version: 8.16.2

### PR DESCRIPTION
**Pull request by childish-sambino**
Fixes #1068

### CODE REVIEW

File: packages/client/src/classes/client.js [hunk 1]
```diff
@@ -15,26 +11,18 @@ const {
...
-   */
   constructor() {
-
-    //API key
     this.apiKey = '';
 
-    //Default headers
     this.defaultHeaders = {
-      'Accept': 'application/json',
+      Accept: 'application/json',
```
Comment (thinkingserious) [inline packages/client/src/classes/client.js]: Why do we convert to a non-string key here? Why only for the `Accept` key?

File: packages/client/src/classes/client.js [hunk 2]
```diff
@@ -43,51 +31,45 @@ class Client {
...
   }
 
-  /**
-   * Set API key
-   */
   setApiKey(apiKey) {
     this.apiKey = apiKey;
+
+    if (!this.isValidApiKey(apiKey)) {
+      console.warn(`API key does not start with ${API_KEY_PREFIX}`);
```
Comment (thinkingserious) [inline packages/client/src/classes/client.js]: Here is a spot to put the message: https://github.com/sendgrid/sendgrid-nodejs/blob/master/packages/helpers/constants/index.js -- nit: period at the end.

File: packages/client/src/classes/client.js [hunk 3]
```diff
@@ -106,52 +88,40 @@ class Client {
...
               return reject(new ResponseError(error.response));
             }
           }
-          // Request error
           return reject(error);
         });
     });
 
-    // Throw and error incase function not passed
+    // Throw an error in case a callback function was not passed/
```
Comment (thinkingserious) [inline packages/client/src/classes/client.js]: nit: period for consistency

File: packages/client/src/classes/client.js (hunk 1 above)
Comment (childish-sambino) [reply packages/client/src/classes/client.js]: Quotes aren't needed here and is an eslint warning.

File: packages/client/src/classes/client.js (hunk 2 above)
Comment (childish-sambino) [reply packages/client/src/classes/client.js]: That's in the helpers package. I don't like defining the message in 1 package and using it in another.

**Pull request merged**